algorithms/src/snark/marlin/ahp/indexer/constraint_system.rs

```diff
@@ -123,7 +141,18 @@ impl<F: Field> CS<F> for ConstraintSystem<F> {
         self.b.push(Self::make_row(&b(LinearCombination::zero())));
         self.c.push(Self::make_row(&c(LinearCombination::zero())));
 
+        self.mul_constraints.insert(self.num_constraints);
+        self.num_constraints += 1;
+    }
+
+    fn lookup(&mut self, _: &[LinearCombination<F>], table_index: usize) -> Result<Variable, SynthesisError> {
```

Comment:
I think the API for this should be similar to the one for `enforce`: we should take in three linear combinations `a, b, c`, and check that the table contains the triple `(az, bz, cz)`. So the API would look like `enforce_lookup(a, b, c)`.

We could additionally keep the existing API of `c = lookup(a, b)` as sugar atop the core API.